lib/CollectionManager.js

```diff
@@ -25,11 +27,28 @@ class CollectionManager {
   }
 
   async dumpCollection() {
-    const query = this.dumpProgress.count
+    const todoQuery = this.dumpProgress.count
       ? { _id: { $gt: this.dumpProgress.token } }
       : {};
-    let cursor = this.collection.find(query);
-    const count = (await cursor.count()) + this.dumpProgress.count;
+    let cursor = this.collection.find(todoQuery).batchSize(BATCH_SIZE);
+    const totalCount = (await cursor.count()) + this.dumpProgress.count;
+
+
+    if (!await cursor.hasNext() && this.dumpProgress.completeDate) {
+      logger.info(`${this.collectionName} already finished on ${this.dumpProgress.completeDate}`);
+      return;
+    }
+
+    if (!await cursor.hasNext() && !this.dumpProgress.completeDate) {
+      await this.dumpProgress.complete();
+      logger.info(`${this.collectionName} has no more to do, marking as finished on ${this.dumpProgress.completeDate}`);
+      return;
+    }
+
+    if (this.dumpProgress.completeDate) {
```

Comment:
Think this will ever happen?

Reply:
Quite possible.  If mongo-stream finishes, then crashes, then more data gets added, and then mongo-stream gets started again. I wanted to make it cover as many cases as possible so that it can be started with the same parameters most of the time without having to worry.